tests/test_benchmark_predictions.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+import pytest
+
+from polaris.evaluate import BenchmarkPredictions
+
+
+def assert_deep_equal(result, expected):
+    assert isinstance(result, type(expected)), f"Types differ: {type(result)} != {type(expected)}"
+
+    if isinstance(expected, dict):
+        assert result.keys() == expected.keys()
+        for key in expected:
+            assert_deep_equal(result[key], expected[key])
+    elif isinstance(expected, np.ndarray):
+        assert np.array_equal(result, expected)
+    else:
+        assert result == expected
+
+
+def test_benchmark_predictions_normalization():
+    # Single task, single test set
+    assert_deep_equal(
+        {"test": {"col1": np.array([1, 2, 3])}},
+        BenchmarkPredictions(
+            predictions=[1, 2, 3], target_labels=["col1"], test_set_labels=["test"]
+        ).predictions,
+    )
+    assert_deep_equal(
+        {"test": {"col1": np.array([1, 2, 3])}},
+        BenchmarkPredictions(
+            predictions={"test": {"col1": [1, 2, 3]}}, target_labels=["col1"], test_set_labels=["test"]
+        ).predictions,
+    )
```

Comment:
Really like that you've put this together in a way where the final form is always accepted as initial input!